homeassistant/components/swiss_public_transport/coordinator.py

```diff
@@ -67,7 +67,7 @@ async def _async_update_data(self) -> DataConnection:
             remaining_time = None
 
         return DataConnection(
-            departure=self._opendata.connections[0]["departure"],
+            departure=departure_time,
             next_departure=self._opendata.connections[1]["departure"],
             next_on_departure=self._opendata.connections[2]["departure"],
```

Comment:
Make use of `parse_datetime()` on these too so we have a proper datetime and then use `isoformat()` to properly convert it to a string